ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1204,90 +1204,104 @@ def install(cls, mh):
         returncode, data = run_healthcheck(
             self.master,
             "ipahealthcheck.ipa.files",
             "IPAFileCheck",
             failures_only=True,
         )
-        assert returncode == 1
-        for check in data:
-            assert check["result"] == "WARNING"
-            assert check["kw"]["key"] == '_etc_resolv.conf_group'
-            assert check["kw"]["type"] == 'group'
-            assert check["kw"]["expected"] == 'root'
-            assert check["kw"]["got"] == 'admins'
-            assert (
-                check["kw"]["msg"]
-                == "Group of %s is admins and should be root"
-                % paths.RESOLV_CONF
-            )
+        if parse_version(version) >= parse_version("0.6"):
+            assert returncode == 1
+            for check in data:
+                assert check["result"] == "WARNING"
+                assert check["kw"]["key"] == "_etc_resolv.conf_group"
+                assert check["kw"]["type"] == "group"
+                assert check["kw"]["expected"] == "root"
+                assert check["kw"]["got"] == "admins"
+                assert (
+                    check["kw"]["msg"]
+                    == "Group of %s is admins and should be root"
+                    % paths.RESOLV_CONF
+                )
+        else:
+            assert returncode == 0
 
     def test_ipa_filecheck_bad_too_restrictive(self, modify_permissions):
+        version = tasks.get_healthcheck_version(self.master)
         modify_permissions(self.master, path=paths.RESOLV_CONF, mode="0400")
         returncode, data = run_healthcheck(
             self.master,
             "ipahealthcheck.ipa.files",
             "IPAFileCheck",
             failures_only=True,
         )
-        assert returncode == 1
-        for check in data:
-            assert check["result"] == "ERROR"
-            assert check["kw"]["key"] == '_etc_resolv.conf_mode'
-            assert check["kw"]["type"] == 'mode'
-            assert check["kw"]["expected"] == '0644'
-            assert check["kw"]["got"] == '0400'
-            assert (
-                check["kw"]["msg"]
-                == "Permissions of %s are too restrictive: "
-                   "0400 and should be 0644"
-                % paths.RESOLV_CONF
-            )
+        if parse_version(version) >= parse_version("0.6"):
+            assert returncode == 1
+            for check in data:
+                assert check["result"] == "ERROR"
+                assert check["kw"]["key"] == "_etc_resolv.conf_mode"
+                assert check["kw"]["type"] == "mode"
+                assert check["kw"]["expected"] == "0644"
+                assert check["kw"]["got"] == "0400"
+                assert (
+                    check["kw"]["msg"]
+                    == "Permissions of %s are too restrictive: "
+                    "0400 and should be 0644" % paths.RESOLV_CONF
+                )
+        else:
+            assert returncode == 0
```

Comment:
For these file permission tests wouldn't it be simpler to skip them if version < 0.6?

Reply:
@rcritten, have added pytest.skip for permission tests.